tools/rest_client.go

```diff
@@ -0,0 +1,97 @@
+package tools
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"net/url"
+	"path"
+	"time"
+
+	"golang.org/x/net/context/ctxhttp"
+)
+
+/*
+ * A rest client to make http calls to datasource rest apis
+ * The client interface allows mocking from unit tests
+ */
+
+var httpClient = &http.Client{
+	Transport: &http.Transport{
+		TLSClientConfig: &tls.Config{
+			Renegotiation: tls.RenegotiateFreelyAsClient,
+		},
+		Proxy: http.ProxyFromEnvironment,
+		Dial: (&net.Dialer{
+			Timeout:   30 * time.Second,
+			KeepAlive: 30 * time.Second,
+			DualStack: true,
+		}).Dial,
+		TLSHandshakeTimeout:   10 * time.Second,
+		ExpectContinueTimeout: 1 * time.Second,
+		MaxIdleConns:          100,
+		IdleConnTimeout:       90 * time.Second,
+	},
+	Timeout: time.Second * 30,
+}
+
+// Client interface
+type Client interface {
+	Fetch(ctx context.Context, uriPath, uriQuery string) ([]byte, error)
+	Get(ctx context.Context, uriPath, uriQuery string) (*http.Response, error)
+}
+
+type restClient struct {
+	url     string
+	headers map[string]string
+}
+
+// NewRestClient ...
+func NewRestClient(url string, headers map[string]string) Client {
+	return &restClient{
+		url,
+		headers,
+	}
+}
+
+// Fetch - perform an HTTP GET and return the body as []byte to prep for marshalling
```

Comment:
Shouldn't we put the documentation comments on the interface methods instead, considering `restClient` isn't used by SDK clients (but instead the interface)?

[Comments documenting declarations should be full sentences](https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences):
```suggestion
// Fetch performs an HTTP GET and returns the body as []byte to prep for marshalling.
```